### PR TITLE
export API for programmatically interfacing with the tree

### PR DESCRIPTION
**Pull request by seiyria**
functions added:
- selectFirst
- selectLast
- selectNext
- selectPrev
- collapse
- expand

This could be used to interface with a hotkey library as thought of in #34.

### CODE REVIEW

**Comment by delbiancodavide** [comment]
doesn't this change break the ability to choose a different name for children's and doubles the amount of data in the scope?

**Comment by seiyria** [comment]
I don't know why it would, but regardless we needed this functionality so I figured I'd submit a patch upstream. I'm in favor of a better way if you've got suggestions.

**Comment by delbiancodavide** [comment]
line 323:
var myIndex = 1 + scope.transcludeScope.$parentNode.children.indexOf(scope.node);

children is hardcoded thus breaking the tree for any datasource that has a different name for the array of children

**Comment by seiyria** [comment]
Ah. Yes, I had to do that because otherwise there was no way to get a hold of the data I needed. Again, I'm all for a better way but I couldn't find one.

**Comment by delbiancodavide** [comment]
I'm trying to figure out one, because i need to make the nodes movable vertically and from their location to the next parent (if there's any) having an interface with arrows would be much easier than drag&drop even if less cool.

**Comment by seiyria** [comment]
I strongly agree. Feel free to let me know if you come up with anything!

**Comment by delbiancodavide** [comment]
It seems like variables are passed thru so to fix the children thing you can change the line 323
from: 
var myIndex = 1 + scope.transcludeScope.$parentNode.children.indexOf(scope.node);
to:
var myIndex = 1 + scope.transcludeScope.$parentNode[scope.$parent.options.nodeChildren].indexOf(scope.node);

File: angular-tree-control.js [hunk 1]
```diff
@@ -269,6 +317,15 @@
                         element.empty();
                         element.append(clone);
                     });
+
+                    var parentIndex = scope.$parent.visibleNodes.indexOf(scope.$parent.node);
+                    if(parentIndex !== -1) {
+                        var myIndex = 1 + scope.transcludeScope.$parentNode.children.indexOf(scope.node);
```
Comment (delbiancodavide) [inline angular-tree-control.js]: should change to:
var myIndex = 1 + scope.transcludeScope.$parentNode[scope.$parent.options.nodeChildren].indexOf(scope.node);
to keep the custom children node feature working

**Comment by 1ftskank** [comment]
Hey, any idea if this will ever get merged? It would be great to have some keyboard navigation.

**Comment by yoavaa** [comment]
This branch has conflicts that need to be resolved, and to make sure all tests are working. 

Also, it does not seem to have any tests that check the functionality introduced in this PR.

**Comment by seiyria** [comment]
sorry, it'll have to be closed then or someone else can pick it up. I stopped working on this in 2014.

**Pull request closed without merging**